File: maui-sample/AccessCode/AccessCodeV1.cs
using System.Globalization;
using System.Security.Cryptography;
using System.Text;

namespace MauiSample.AccessCode;

public static class AccessCodeV1
{
  public static string Generate(Guid appID, DateTime utcTime)
  {
    string lowercaseID = appID.ToString("D");
    // Format utcTime as an ISO8601 compliant string, like this: 2024-02-22T18:00:00Z
    string iso8601Time = utcTime.ToString("yyyy-MM-dd'T'HH:mm:ssK", CultureInfo.InvariantCulture);
    string plaintextAccessCode = lowercaseID + iso8601Time;
    byte[] utf8Bytes = Encoding.UTF8.GetBytes(plaintextAccessCode);
    byte[] hashedBytes = SHA256.HashData(utf8Bytes);
    string base64String = Convert.ToBase64String(hashedBytes);
    return base64String;
  }
}

File: android-native-sample/app/src/main/java/com/example/kotlin_sample/accesscode/AccessCodeV1.kt
package com.example.kotlin_sample.accesscode

import java.security.MessageDigest
import java.time.Instant
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter
import java.util.Base64
import java.util.UUID

object AccessCodeV1 {
  private val iso8601Formatter: DateTimeFormatter =
    DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ssX")
      .withZone(ZoneOffset.UTC)

  fun generate(appId: UUID, utcTime: Instant): String {
    val lowercaseId = appId.toString().lowercase()
    val iso8601Time = iso8601Formatter.format(utcTime)
    val plaintextAccessCode = lowercaseId + iso8601Time
    val hashedBytes = MessageDigest.getInstance("SHA-256")
      .digest(plaintextAccessCode.toByteArray(Charsets.UTF_8))
    return Base64.getEncoder().encodeToString(hashedBytes)
  }
}
